# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/nsWifiAccessPoint.h
# Commit: 028af3d71e19
# Full Hash: 028af3d71e1918406a0bc3b249ef8feaa0348643
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-13 21:42:25
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Only scan wifi when the network changes or when on mobile r=necko-reviewers,kershaw
#   
#   Simplifies the concurrent operations of wifi scanning and reduces the
#   frequency in common cases.  Wifi scanning when on mobile is reduced from
#   every 5 seconds to every minute.  Wifi scans will also happen whenever
# ==============================================================================

diff -r 35c408b9531d -r 028af3d71e19 netwerk/wifi/nsWifiAccessPoint.h
--- a/netwerk/wifi/nsWifiAccessPoint.h	Sat May 13 00:36:00 2023 +0000
+++ b/netwerk/wifi/nsWifiAccessPoint.h	Sat May 13 00:36:01 2023 +0000
@@ -67,13 +67,12 @@
       mSsidLen = 0;
     }
   }
+
+  // 3-value compare for nsWifiAccessPoint
+  int Compare(const nsWifiAccessPoint& o) const;
+
+  bool operator==(const nsWifiAccessPoint& o) const;
+  bool operator!=(const nsWifiAccessPoint& o) const { return !(*this == o); }
 };
 
-// Helper functions
-
-bool AccessPointsEqual(nsCOMArray<nsWifiAccessPoint>& a,
-                       nsCOMArray<nsWifiAccessPoint>& b);
-void ReplaceArray(nsCOMArray<nsWifiAccessPoint>& a,
-                  nsCOMArray<nsWifiAccessPoint>& b);
-
 #endif